syncer/syncer.go

```diff
@@ -2122,7 +2122,7 @@ func (s *Syncer) handleRowsEvent(ev *replication.RowsEvent, ec eventContext) err
 			ec.tctx.L().Info("re-replicate shard group was completed", zap.String("event", "row"), zap.Stringer("re-shard", ec.shardingReSync))
 			return ec.closeShardingResync()
 		}
-		if ec.shardingReSync.targetSchema != schemaName || ec.shardingReSync.targetTable != tableName {
+		if !reflect.DeepEqual(ec.shardingReSync.targetTable, &filter.Table{Schema: schemaName, Name: tableName}) {
```

Comment:
perfer not to use reflect here for better performance, maybe use `ec.shardingReSync.targetTable.String()==&filter.Table{Schema: schemaName, Name: tableName}.String()`